src/ur/workday.ur

```diff
@@ -0,0 +1,378 @@
+     Href : string
+     AlsoAbout : list descriptor, (* other workers the feedback is about *)
+     RelatedFeedbackEvents : list relatedFeedback,
+     RelatesTo : descriptor,
+      *)
+}
+val json_anytimeFeedback : json anytimeFeedback =
+    json_record_withOptional
+        {Id = "id"}
+        {Preview = "descriptor",
+         ToWorker = "toWorker",
+         WorkersToNotify = "workersToNotify",
+         BusinessProcessParameters = "businessProcessParameters",
+         HiddenFromWorker = "hiddenFromWorker",
+         HiddenFromManager = "hiddenFromManager",
+         Comment = "comment",
+         FromWorker = "fromWorker",
+         OverallStatus = "overallStatus",
+         GivenDate = "feedbackGivenDate",
+         ShowFeedbackProviderName = "showFeedbackProviderName"
+                 (*
+                 RelatesTo = "relatesTo",
+                 Badge = "badge",
+                 AlsoAbout = "feedbackAlsoAbout",
+                 RelatedFeedbackEvents = "relatedFeedbackEvents",
+                  *)
+                }
+
+datatype feedbackStatus = Complete | Requested
+(*
+val show_feedbackStatus : show feedbackStatus =
+ fn s => case s of
+             Complete => "Complete"
+           | Requested => "Requested"
+val read_feedbackStatus : read feedbackStatus
+val eq_feedbackStatus : eq feedbackStatus
+*)
+
+con feedback = [
+      Id = wid,
+      Created = time,
+      RequestedDate = option time,
+      About = wid, (* who the feedback is about *)
+      Provider = wid, (* who writes the feedback *)
+      RequestedBy = option wid,
+      Body = string, (* response *)
+      Guidance = string,
+      Status = string, (* TODO: use feedbackStatus *)
+      SendMail = bool,
+      LastUpdated = time
+]
+
+con feedbackWithoutId = [
+      Created = time,
+      RequestedDate = option time,
+      About = wid, (* who the feedback is about *)
+      Provider = wid, (* who writes the feedback *)
+      RequestedBy = option wid,
+      Body = string, (* response *)
+      Guidance = string,
+      Status = string, (* TODO: use feedbackStatus *)
+      SendMail = bool,
+      LastUpdated = time
+]
+
+(* ************************************************************************** *)
+
+functor Make(M : AUTH) = struct
+    open M
+
+    (*
+     desired features:
+      * allow manager to request feedback of their subordinates on each other
+      * notify the subordinate of this request (by email)
+      * allow the subordinate to submit the feedback
+      * notify the manager when feedback is submitted
+      * allow manager to see the feedbacks they've requested
+      * allow the subordinate to see the feedback they've been assigned
+     *)
+
+
+    val token =
+        toko <- token;
+        case toko of
+            None => error <xml>You must be logged into Workday to use this feature.</xml>
+          | Some tok => return tok
+
+    val prefixCommon = "https://wcpdev-services1.wd101.myworkday.com/ccx/api/v1/janestreet_wcpdev1"
```

Comment:
Hm, we might even need to scrub Git history before pushing to the public World repo.  I don't know if these URL prefixes are considered authentication data or otherwise sensitive information by Jane Street.  Check out how instead this information is provided from Nectry for Salesforce, asking me if you have questions on how it works.  I acknowledge it's a bit twisty, but also the ideas we have for overhauling API categories should allow for a nicer flow.

Reply:
Certainly. We can interactively rebase at a later date. I made a task for this in our Projects board.

The Salesforce implementation using API Parameters is quite clear to me; adapting this implementation now.